### DISCUSSION TITLE
[Faucet] Large amounts of CELO/cUSD/cEUR/cREAL on Alfajores

### DISCUSSION THREAD

**Discussion by arthurgousset** [celo-org/faucet#43] [Q&A] [answered]
Hey all, we've been getting frequent requests for larger amounts of CELO/cUSD/cEUR/cREAL on Alfajores. 

E.g.:
- > [...] is there a way to receive more CELO on Alfajores? We are building defi protocol hence for testnet testing we would need a larger amount of CELO, given CELO usd amount < $1 it's rather hard to test positions etc without having a big testnet CELO stash ~ [Discord user](https://discord.com/channels/600834479145353243/774302385555243008/1031524307601461338)
- > [...] is there a way to get more cUSD on alfajores than using the faucet? ~ Private DM

@dckesler and I were speaking about this last week and are brainstorming ideas.

If this is something you need as a high priority, please upvote this post and/or let me know about your specific needs below.
Thanks in advance!

**Comment by fbaud** [comment]
Hi Arthur,

1 Celo is usually ok in my context, but 1 unit for each stable coin is certainly too low. It used to be 10, and it was ok to invite people to test an app and ask them to fund their Alfajores account prior to the test and spend a couple of cUSD or cEUR in a mockup scenario. Asking to spend a couple of cents is making demos less realistic.

So I'm in favour of raising back the amount of stable coins to 10 (if possible, or even higher depending on the constraints).

**Reply by arthurgousset** [reply]
Thanks for sharing @fbaud 💯

**Comment by vibern0** [comment]
10 cUSD used to be fine. We use the funds to test impactMarket. We always try to circulate all the money (aka, receive UBI and donate back again) and because UBIs are usually low, low amounts is fine. But now with just 1 cUSD is very hard and, as we move forward beyond UBI and also would be great to test with larger amounts it would be perfect if we can request a larger amount. Likely, a one time large amount is enough.

**Reply by arthurgousset** [reply]
Thanks for sharing @obernardovieira !

**Comment by arthurgousset** [comment]
(_sharing for visibility_) 
Today, I came across the [MultiFaucet](https://faucet.paradigm.xyz/) by Paradigm.
This might be a useful inspiration for Celo's own large Alfajores faucet.
It uses a Twitter account to protect from bot attacks:

> Your Twitter account must have at least 1 Tweet, 15 followers, and be older than 1 month.
> By default, the faucet drips on the Ethereum testnets (Görli). You can choose to receive a drip on other networks when requesting tokens.
> You can claim from the faucet once every 24 hours

<img width="500" alt="image" src="https://user-images.githubusercontent.com/46296830/197196160-427fff23-731f-4738-a043-ce61414976d1.png">

**Reply by fbaud** [reply]
Maybe we should define different categories of users:
- for developpers, twitter, bots on a discord channel or any multi-step operations is ok and could serve to distribute larger amounts
- for ordinary users, that just need a one-off amount to test an app, it should be very simple and I expect posting a Tweet as a show-stopper

**Reply by arthurgousset** [reply]
> for ordinary users, that just need a one-off amount to test an app, it should be very simple and I expect posting a Tweet as a show-stopper

Agreed, that's why the [existing Faucet](https://celo.org/developers/faucet) is not protected.

This discussion is specifically for large CELO requests, which need to be protected somehow 👍

**Reply by fbaud** [reply]
@0xarthurxyz would be nice if the allocation 1cUSD/cEUR/cReal could be raised back to 10 for ordinary users.

**Reply by arthurgousset** [reply]
Noted ✅

**Comment by arthurgousset** [comment]
(_sharing for visibility_)
Came across Alchemy's [faucet for the Polygon Mumbai testnet](https://www.alchemy.com/overviews/mumbai-faucet).

<img width="500" alt="image" src="https://user-images.githubusercontent.com/46296830/199743163-8ce712d7-a1ed-4b0a-9335-365963745fca.png">

Posting here for future reference, as we might be able to learn a thing or two.

**Comment by arthurgousset** [comment]
(_sharing for visibility_)
This is how Mudit Gupta does it for Goerli ETH: https://goerli-faucet.mudit.blog/

> To prevent malicious actors from exhausting all available funds or accumulating enough Ether to mount long running spam attacks, requests are tied to common 3rd party social network accounts. Anyone having a Twitter or Facebook account may request funds within the permitted limits.

<img width="500" alt="image" src="https://user-images.githubusercontent.com/46296830/204393541-b0a15e10-e682-42d7-bea8-7790c0949175.png">

**Comment by arthurgousset** [comment]
(_sharing for visibility_)
Here is how Quicknode does it for Goerli ETH: https://faucet.quicknode.com/ethereum/goerli

<img width="400" alt="image" src="https://user-images.githubusercontent.com/46296830/204394207-8bcf0947-ca29-481e-9113-9c5c39940e91.png">

<img width="400" alt="image" src="https://user-images.githubusercontent.com/46296830/204394304-055bfa31-dc5b-463d-94ce-795b9cfa4173.png">

**Comment by ohenepee** [comment]
Seriously, what the hell is Celo.org protecting? Server resources or what exactly? Here's the case whereby I need roughly a 1000 "fake" native CELO to test my contract and I'm torn between tapping the faucet for days or moving to another blockchain where I can actually get large amounts of their testnet coins. What's the point of moving from third world country to country engaging crypto enthusiasts and devs to jump unto your blockchain if you're gonna add to their already hellish nightmare of slow internet connections? It already takes 2 mins to get 1 CELO from my end.

**Reply by arthurgousset** [reply]
Thank for your feedback. You can request larger amount of testnet tokens using this form: [Faucet: Request large amounts of testnet tokens](https://forms.gle/V5zKpdZkAumLpj6T8)

The reason proof-of-stake L1s have to perform sybil resistance checks on testnet token requests is to avoid a situation in which a malicious actor self elects validators and takes over the testnet.

This is an issue across many ETH testnets as you can see from the various solutions documented here.

**Comment by arthurgousset** [comment]
(_sharing for future reference_)
A16Z's sybil resistant Faucet design: https://github.com/a16z/eth-testnet-drop

<img width="800" alt="image" src="https://user-images.githubusercontent.com/46296830/206039495-b1e089df-c1e0-4d7a-9231-354adbf4444c.png">

**Comment by aaronmgdr** [comment, marked as answer]
### A Few  Faucet Updates. 

#### For the faucet at [faucet.celo.org](https://faucet.celo.org)

1.  changed to give out 5 of each token. 
2. now receives tokens daily so if its out it will automatically have more the next day
3. if any stable token runs out it will convert CELO to that token using Exchange contract before sending

### For Larger Faucet needs

* The Celo Foundation DevRel Team is now using a Celo Safe to fulfill requests this means more people are able to do the disbursing. still using the same [form](https://forms.gle/V5zKpdZkAumLpj6T8)



### Future Considerations 

We are Not necessarily doing these yet might. 

* on the little faucet make each token opt in 
* on the little faucet giving choice of 1, 5, or 10 tokens

The Idea with these is to limit sending out tokens that are not needed or wanted and at the same time increase the amounts of those that are wanted when needed, 

* create a system for more easily getting between 10 and 100 tokens.

**Reply by arthurgousset** [reply]
Thanks for the detailed write up @aaronmgdr !

**Comment by ohenepee** [comment]
I filled the form over 3 weeks ago for larger amounts of CELO and cUSD, and I've received nothing, absolutely nothing!

When I visit the link, I'm told I've already responded which means I can't refill the form.

**Reply by aaronmgdr** [reply]
@nestorbonilla

**Reply by nestorbonilla** [reply]
Hi @ohenepee, could you share with us from which Twitter username and/or the public key did you request the funds? I don't see any Twitter username similar to your GitHub username on our form request records.

**Reply by ohenepee** [reply]
Public Key: 0x5e07CC37a69e30b45096032496674A40671ba384

**Comment by aaronmgdr** [comment]
btw same backend but there is a new UI for the faucet at faucet.celo.org
